cairo/tests/utils/args_gen.py

```diff
@@ -558,15 +574,38 @@ def _gen_arg(
             for f in fields(arg_type_origin)
         ]
         segments.load_data(struct_ptr, data)
+
+        if arg_type_origin is Trie:
+            # In case of a Trie, we need the dict to be a defaultdict with the trie.default as the default value.
+            dict_ptr = segments.memory[data[2]]
+            default_arg = _gen_arg(
+                dict_manager, segments, get_args(arg_type)[1], arg.default
+            )
+            dict_manager.trackers[dict_ptr.segment_index].data = defaultdict(
+                lambda: default_arg, dict_manager.trackers[dict_ptr.segment_index].data
+            )
+
         return struct_ptr
 
     if arg_type in (U256, Hash32, Bytes32, Bytes256):
         if isinstance_with_generic(arg, U256):
             arg = arg.to_be_bytes32()[::-1]
+
+        felt_values = [
+            int.from_bytes(arg[i : i + 16], "little") for i in range(0, len(arg), 16)
+        ]
+
+        if hash_mode:
+            if len(felt_values) != 2:
+                raise ValueError(
+                    "Hashmode is currently only supported for 32-bytes integers"
+                )
+            return poseidon_hash(felt_values[0], felt_values[1])
+
         base = segments.add()
         segments.load_data(
             base,
-            [int.from_bytes(arg[i : i + 16], "little") for i in range(0, len(arg), 16)],
+            felt_values,
```

Comment:
remove trailing comma, trailing commas are a special case to enforce line breaking for each variable, and it'll be on the same line

Reply:
no use case found yet

Reply:
done